server/src/parse_thread.rs

```diff
@@ -81,6 +80,53 @@ impl File {
     }
 }
 
+struct DiagnosticHandler<'a> {
+    output: &'a tokio::sync::mpsc::UnboundedSender<ParserMsg>,
+}
+
+impl<'a> DiagnosticHandler<'a> {
+    fn handle_spanned<T: Spanned>(
+        &self,
+        x: &[T],
+        severity: lsp::DiagnosticSeverity,
+        url: &lsp::Url,
+        file: &File,
+    ) {
+        self.output
```

Comment:
this sort of allocation free, emitting of diagnostics as they arrive doesn't really work,
because the editor is stateful, so emitting a diagnostic has the effect of clearing previous diagnostics for the same file.

As such we need to store these somewhere until we are done.